src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Android.cs

```diff
@@ -28,7 +28,7 @@ internal partial class ModalNavigationManager
 		AAnimation? _dismissAnimation;
 		bool _platformActivated;
 
-		readonly Stack<string> _modals = [];
+		readonly List<string> _modals = [];
```

Comment:
it looks good to me. Just a heads up, Lists aren't good to remove at start, since it will cause the whole collection to reorganize. So, if in the future, we see some scenario where this can cause a bottleneck, we can change this to other collection.

I doubt it will happen but just in case